integration/main_test.go

```diff
@@ -28,6 +27,14 @@ func TestMain(m *testing.M) {
 		dbTest = *db
 	}
 
+	// start the postgres container; stop container when finished
+	log.Printf("running the test postgres container")
+	closeDatabase, err := dbTest.RunAsDockerContainer("contacts-application-test")
+	if err != nil {
+		log.Fatalf("failed to launch the test postgres container: %v", err)
+	}
+	defer closeDatabase()
```

Comment:
> I also think fatal will stop deferred functions from executing, which (I think) would keep an unused database connection alive.

I am a terrible hypocrite. Definitely worth revisiting a better cleanup process when database stuff gets moved to the toolkit (that's my bad, not yours).

Reply:
Spent some time Googling-around. It's possible to gracefully exit a Go program and still run the defer functions. It would be a matter of rewriting the `log.Fatalf` logs to `log.Errorf` logs and calling `runtime.Goexit()` in the same error check. 

Again, this is something I can change because it's not completely related to this pull request.

Here's a [good source](https://stackoverflow.com/questions/27629380/how-to-exit-a-go-program-honoring-deferred-calls). Bonus: I'll share [this related post](https://groups.google.com/forum/#!topic/golang-nuts/UYoaWk6oJbY) because the back-and-forth is really entertaining.